CodeStarterCamp_Week3/main.swift

```diff
@@ -12,55 +12,88 @@ enum Coffee {
-            if buyAvailability {
-                print("\(self.name) paid \(cost * quantity) for \(quantity) \(productName)(s)")
-            } else {
-                print("you do not have enough money")
-            }
+        return buyAvailability
+    }
+    
+    mutating func printPurchaseResult (buyAvailability: Bool, productName: Coffee) {
+        if buyAvailability {
+            self.money -= productName.cost
+            print("\(self.name) paid \(productName.cost) for \(productName)")
+        } else {
+            print("잔액이 {\((productName.cost) - self.money)}원만큼 부족합니다.")
         }
+    }
 }
 
 struct CoffeeShop {
     var revenue: Int
     let menu: [Coffee: Int]
-    let barista: Person
-    var pickUpTable: [String: String]
+    private let barista: Person
+    private var pickUpTable: Array<String> {
+        willSet (newName) {
+            guard let personName = newName.first else { return }
+            print("\(personName) 님의 커피가 준비되었습니다. 픽업대에서 갖져가 주세요.")
+        }
+    }
     
-    mutating func takeOrder(person: Person, coffee: Coffee) {
+    init(revenue: Int, menu: [Coffee: Int], barista: Person, pickUpTable: Array<String>) {
+        self.revenue = revenue
+        self.menu = menu
+        self.barista = barista
+        self.pickUpTable = pickUpTable
+    }
+    
+    mutating func canOrder(coffee: Coffee) -> Bool {
         var orderAvailability = false
-        if let coffeePrice = menu[coffee] {
-            revenue += coffeePrice
+        if let _ = menu[coffee] {
             orderAvailability = true
         }
-        makeCoffee(orderAvailability: orderAvailability, person: person, coffee: coffee)
+        return orderAvailability
     }
-    func makeCoffee(orderAvailability: Bool, person: Person, coffee: Coffee) {
-        if orderAvailability {
-            print("we made \(coffee) for \(person.name)")
-        } else {
-            print("We don't have \(coffee) in our store")
-        }
-        
+    
+    mutating func makeCoffee(person: Person, coffee: Coffee) {
+        self.revenue += coffee.cost
+        pickUpTable.append(person.name)
     }
 }
 
-var misterLee = Person(name: "misterLee", money: 10000)
-var missKim = Person(name: "missKim", money: 20000)
+var misterLee = Person(name: "misterLee", money: 5000)
+var missKim = Person(name: "missKim", money: 10000)
 
-var yagomBucks = CoffeeShop(revenue: 0, menu: [Coffee.americano: 4500, Coffee.latte: 5000, Coffee.frappuccino: 5500], barista: misterLee, pickUpTable: [:])
+var yagomBucks = CoffeeShop(revenue: 0, menu: [Coffee.americano: Coffee.americano.cost, Coffee.latte: Coffee.latte.cost, Coffee.frappuccino: Coffee.frappuccino.cost], barista: misterLee, pickUpTable: [])
 
+if missKim.canBuy(productName: Coffee.americano) {
+    if yagomBucks.canOrder(coffee: .americano) {
+        missKim.printPurchaseResult(buyAvailability: true, productName: .americano)
+        yagomBucks.makeCoffee(person: missKim, coffee: .americano)
+    }
```

Comment:
if 문이 중첩되네요! 아래와 같이 바꿀 수 있습니다.
```suggestion
if missKim.canBuy(productName: Coffee.americano),
   yagomBucks.canOrder(coffee: .americano) {
        missKim.printPurchaseResult(buyAvailability: true, productName: .americano)
        yagomBucks.makeCoffee(person: missKim, coffee: .americano)
    }
```

Reply:
새로운 방법을 알려주셔서 감사합니다!

Reply:
https://github.com/yagom-academy/swift-starter-week3/pull/47/commits/1d16b51961f5300dbe6e4817de0c180b78d2caae

이부분은 차후에 중첩if문을 사용할 때 유용하게 사용하겠습니다!